metricflow/model/validations/validator_helpers.py

```diff
@@ -39,82 +40,117 @@ class ModelObjectType(Enum):
     METRIC = "metric"
 
 
+class ValidationContext(BaseModel):
+    """The base context class for vaidation issues"""
+
+    file_name: Optional[str]
+    line_number: Optional[int]
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+
+        context_string = ""
+
+        if self.file_name:
+            context_string += f"in file `{self.file_name}`"
+            if self.line_number:
+                context_string += f" on line #{self.line_number}"
+
+        return context_string
+
+
+class MaterializationContext(ValidationContext):
+    """The context class for vaidation issues involving materializations"""
+
+    materialization_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with materialization `{self.materialization_name}` {ValidationContext.context_str(self)}"
+
+
+class MetricContext(ValidationContext):
+    """The context class for vaidation issues involving metrics"""
+
+    metric_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with metric `{self.metric_name}` {ValidationContext.context_str(self)}"
+
+
+class DataSourceContext(ValidationContext):
+    """The context class for vaidation issues involving data sources"""
+
+    data_source_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with data source `{self.data_source_name}` {ValidationContext.context_str(self)}"
+
+
+class DimensionContext(DataSourceContext):
+    """The context class for vaidation issues involving dimensions"""
+
+    dimension_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with dimension `{self.dimension_name}` in data source `{self.data_source_name}` {ValidationContext.context_str(self)}"
+
+
+class IdentifierContext(DataSourceContext):
+    """The context class for vaidation issues involving indentifiers"""
+
+    identifier_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with identifier `{self.identifier_name}` in data source `{self.data_source_name}` {ValidationContext.context_str(self)}"
+
+
+class MeasureContext(DataSourceContext):
+    """The context class for vaidation issues involving measures"""
+
+    measure_name: str
+
+    def context_str(self) -> str:
+        """Human readable stringified representation of the context"""
+        return f"with measure `{self.measure_name}` in data source `{self.data_source_name}` {ValidationContext.context_str(self)}"
+
+
+ContextType = Union[
```

Comment:
Do we need this? Everything inherits from ValidationContext, so we could just use that.

The other alternative, if you want to differentiate between a concrete ValidationContext and a general validation context object, is to define this as a Protocol instead of a custom union type, which feels very clumsy.

Note this is mainly an aesthetic comment. I don't think it makes much practical difference and it is easy enough to change.